### PR TITLE
feat: adjust padding when using filters in table

### LINKED ISSUE #3231: [Table] Filterable column header padding adjustment
### What package is this for?

- [x] React
- [ ] Angular

### Summary

The padding around the filterable column headers has been adjusted to try and give the most space possible to the text field, while at the same time remaining visually separated from column-to-column. This has also been adjusted with better consideration to filterable right-aligned columns and selectable + filterable columns.

#### Proposed styling
Screen Shot 2022-01-26 at 10 44 41 AM

#### Red lines & interaction states
image

image

### PR DESCRIPTION
**Pull request by kevinsperrine**
Closes #3231 

**Summary**

This isn't a full 100% fix. The right-align items specifically were left out, because we don't currently change the alignment of the headers--only the column content when aligning right or center. Otherwise, the other items were addressed, I believe.

**Change List (commits, features, bugs, etc)**

- it adds padding to the edges and bottom of the filters
- adjusts the height of the header based on table row size
- adjusts padding to align column title with edge of content/placeholder in filters
- added borders between filter sections when using column groupings
- updated snapshots

**Acceptance Test (how to verify the PR)**

- confirm these changes match the styles in the original issue--minus the right-alignment specs

**Regression Test (how to make sure this PR doesn't break old functionality)**

- check RTL/LTR with various knobs and demos in the playground to make sure edge cases weren't missed

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by bjornalm** [comment]
image

The sort button in RTL

**Review by bjornalm — CHANGES_REQUESTED** [review]
This is a bit difficult to review since many details of our columns are still are different from the design specs attached to the issue. But looking at the change list of the PR I do find some new issues:

The last column's filter cell has a white border to right right
image

and in RTL it looks really strange:
image

For the XL size column group text is missing top padding/margin
image

The text of the input is now aligned with thet title of the column, which I think looks great, but it is not according to the specs I think

File: packages/react/src/components/Table/TableHead/FilterHeaderRow/_filter-header-row.scss [hunk 1]
```diff
@@ -15,9 +16,56 @@
...
 }
 
+.#{$prefix}--data-table
+  th.#{$iot-prefix}--filter-header-row--header.#{$iot-prefix}--tableheader-filter {
+  padding-right: $spacing-02;
+  padding-left: $spacing-02;
+  padding-bottom: $spacing-03;
+
+  &.#{$iot-prefix}--tableheader-filter:not(:first-of-type) {
+    border-left-width: 2px;
```
Comment (davidicus) [inline packages/react/src/components/Table/TableHead/FilterHeaderRow/_filter-header-row.scss:26]: Should these be abstracted into a variable since we use it in a few places?

**Review by davidicus — APPROVED** [review]
LGTM 👍 Just one comment about possible variable but not a reason to hold up.

File: packages/react/src/components/Table/TableHead/FilterHeaderRow/_filter-header-row.scss (hunk 1 above)
Comment (kevinsperrine) [reply packages/react/src/components/Table/TableHead/FilterHeaderRow/_filter-header-row.scss]: added a variable.
